lib/connextclient/ConnextClient.ts

```diff
@@ -0,0 +1,513 @@
+      // we are the maker paying the taker
+      amount = deal.takerUnits;
+      tokenAddress = this.tokenAddresses.get(deal.takerCurrency)!;
+    } else {
+      // we are the taker paying the maker
+      amount = deal.makerUnits;
+      tokenAddress = this.tokenAddresses.get(deal.makerCurrency)!;
+      lock_timeout = deal.makerCltvDelta!;
+    }
+
+    try {
+      const secret = await this.executePaymentTransfer({
+        amount,
+        lock_timeout,
+        token_address: tokenAddress,
+        target_address: deal.destination!,
+        secret_hash: deal.rHash,
+      });
+      return secret;
+    } catch (err) {
+      switch (err.code) {
+        case 'ECONNRESET':
+        case errorCodes.UNEXPECTED:
+        case errorCodes.TIMEOUT:
+        case errorCodes.SERVER_ERROR:
+        case errorCodes.INVALID_TOKEN_PAYMENT_RESPONSE:
+          throw swapErrors.UNKNOWN_PAYMENT_ERROR(err.message);
+        default:
+          throw swapErrors.FINAL_PAYMENT_ERROR(err.message);
+      }
+    }
+  }
+
+  public addInvoice = async () => {
+    // not implemented, connext does not use invoices
+  }
+
+  public settleInvoice = async () => {
+    // not implemented, connext does not use invoices
+  }
+
+  public removeInvoice = async () => {
+    // not implemented, connext does not use invoices
+  }
+
+  public lookupPayment = async (
+    rHash: string,
+    currency?: string,
+    destination?: string,
+  ) => {
+    // TODO: change identifier
+    const identifier = 0;
+
+    // first check if the payment is pending
+    const pendingTransfers = await this.getPendingTransfers(
+      currency,
+      destination,
+    );
+    for (const pendingTransfer of pendingTransfers) {
+      if (identifier === Number(pendingTransfer.payment_identifier)) {
+        return { state: PaymentState.Pending };
+      }
+    }
+
+    // if the payment isn't pending, check if it has succeeded or failed
+    const paymentEvents = await this.getPaymentEvents(currency, destination);
+    for (const paymentEvent of paymentEvents) {
+      if (paymentEvent.identifier === identifier) {
+        const success = paymentEvent.event === 'EventPaymentSentSuccess';
+        if (success) {
+          const preimage = paymentEvent.secret;
+          return { preimage, state: PaymentState.Succeeded };
+        } else {
+          return { state: PaymentState.Failed };
+        }
+      }
+    }
+
+    // if there is no pending payment or event found, we assume that the payment was never attempted by connext
+    return { state: PaymentState.Failed };
+  }
+
+  private getPendingTransfers = async (
+    currency?: string,
+    destination?: string,
+  ) => {
+    // const channel = this.getConnextChannel();
+    // TODO: ndeed to expose from node api
+    const res = this.channel?.isAvailable();
+    return (res as any);
+  }
+
+  private getPaymentEvents = async (
+    currency?: string,
+    destination?: string,
+  ) => {
+    let endpoint = 'payments';
```

Comment:
Lots of unused `endpoint` variables in this file. I'm assuming it's a copy paste leftover from `RaidenClient` and won't comment on it further.